src/test/java/karate-config.js

```diff
@@ -15,6 +15,7 @@ function fn() {
 
   switch (env) {
     case 'dev':
+        config.urlNodeJS = 'https://nodejs-qa-training.herokuapp.com';
```

Comment:
Maybe you could put the environment variables in an env_data.json file and let karate-config.js read that file, this way you avoid changing the .js every time you upload data from an environment